packages/core/src/context/starknet.tsx

```diff
@@ -174,6 +176,21 @@ function useStarknetManager({
     [updateChainAndProvider, state.currentProvider],
   );
 
+  const handleDefaultChainChange = useCallback(() => {
+    if (!connectorRef.current) {
+      // Only update currentChain if no wallet is connected
+      setState((state) => ({
+        ...state,
+        currentChain: defaultChain,
+        currentProvider: providerForChain(defaultChain, provider).provider,
+      }));
+    }
+  }, [defaultChain, provider]);
+
+  useEffect(() => {
+    handleDefaultChainChange();
+  }, [handleDefaultChainChange]);
+
```

Comment:
```suggestion
  useEffect(() => {
    if (!connectorRef.current) {
      // Only update currentChain if no wallet is connected
      setState((state) => ({
        ...state,
        currentChain: defaultChain,
        currentProvider: providerForChain(defaultChain, provider).provider,
      }));
    }
  }, [defaultChain, connectorRef, provider]);
```

I believe it can be cleaned up to only use useEffect. This way it also should handle when the user disconnects the wallet.